frontend/src/metabase/query_builder/components/notebook/steps/JoinStep/JoinConditionColumnPicker/JoinConditionColumnPicker.tsx

```diff
@@ -1,143 +1,190 @@
-import { forwardRef, useCallback, useMemo } from "react";
+import type { Ref } from "react";
+import { forwardRef, useMemo } from "react";
 import { t } from "ttag";
 
-import type {
-  QueryColumnPickerProps,
-  ColumnListItem,
-} from "metabase/common/components/QueryColumnPicker";
+import type { ColumnListItem } from "metabase/common/components/QueryColumnPicker";
 import { Popover, Text } from "metabase/ui";
 import * as Lib from "metabase-lib";
 
 import {
-  JoinConditionCellItem,
-  StyledQueryColumnPicker,
+  JoinCellItem,
+  JoinColumnPicker,
 } from "./JoinConditionColumnPicker.styled";
 
-interface JoinConditionColumnPickerProps
-  extends Omit<QueryColumnPickerProps, "checkIsColumnSelected"> {
-  isNewCondition: boolean;
-  column?: Lib.ColumnMetadata;
-  table?: Lib.Joinable;
-  label?: string;
+interface JoinConditionColumnPickerProps {
+  query: Lib.Query;
+  stageIndex: number;
+  joinable: Lib.Join | Lib.Joinable;
```

Comment:
We have `JoinOrJoinable` that could be used here and in `JoinColumnDropdownProps`, it just need to be exported.

```suggestion
  joinable: Lib.JoinOrJoinable;
```

Reply:
Fixed